lib/service/Service.ts

```diff
@@ -33,18 +37,22 @@ class Service {
    * Get general information about walli-server and the nodes it is connected to
    */
   public getInfo = async (): Promise<WalliInfo> => {
-    const { btcdClient, lndClient, ltcdClient } = this.serviceComponents;
+    const { btcdClient, ltcdClient, lndClient, xudClient } = this.serviceComponents;
     const version = packageJson.version;
 
-    const btcdInfo = await btcdClient.getInfo();
-    const ltcdInfo = await ltcdClient.getInfo();
-    const lndInfo = await lndClient.getLndInfo();
+    const info = await Promise.all([
+      btcdClient.getInfo(),
+      ltcdClient.getInfo(),
+      lndClient.getLndInfo(),
+      xudClient.getXudInfo(),
+    ]);
 
     return {
       version,
-      btcdInfo,
-      ltcdInfo,
-      lndInfo,
+      btcdInfo: info[0],
```

Comment:
Yeah this is really ugly but I can't come with a better solution right now

Reply:
I will think of a more aesthetic way, in the meanwhile, I marked it with a TODO.